app/controllers/api/v1/interactive_pages_controller.rb

```diff
@@ -76,7 +86,7 @@ def set_sections
     # Remove deleted sections:
     old_sections.each do |section|
       unless (new_section_ids.include?(section.id.to_s))
-        section.delete
+        section.update_attribute(:interactive_page_id, nil)
```

Comment:
Sorry but I don't quite understand this change. I understand that it's setting the section's interactive_page_id value to nil and that that must negate its association with the page. But I don't understand why that's an improvement over `section.delete`. Do the deleted sections need to kept around for some reason even though they're no longer associated with the page? Is this related to the handling of moving a section between pages? I'm guessing so, but it's not quite clear to me.

Reply:
When we 'move' a section, we just remove its association from the page where it is now, and add a new association to its destination. We update each page separately. 

Before this change, we would lose sections when moving them in LARA sometimes.